velox/functions/sparksql/String.h

```diff
@@ -845,4 +845,103 @@ struct TranslateFunction {
   }
 };
 
+/// Decodes `string` in 'application/x-www-form-urlencoded' format using UTF-8
+/// encoding scheme.
+template <typename T>
+struct UrlDecoderFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // ASCII input does NOT always produce ASCII result.
+  static constexpr bool is_default_ascii_behavior = false;
+
+  int decode(StringView input) {
+    assert(input.size() == 2);
+    auto toIntFromHex = [](char x) {
+      switch (x) {
+        case '0':
+          return 0;
+        case '1':
+          return 1;
+        case '2':
+          return 2;
+        case '3':
+          return 3;
+        case '4':
+          return 4;
+        case '5':
+          return 5;
+        case '6':
+          return 6;
+        case '7':
+          return 7;
+        case '8':
+          return 8;
+        case '9':
+          return 9;
+        case 'a':
+        case 'A':
+          return 10;
```

Comment:
The behavior is case-insensitive, is that intentional?

Reply:
Confirmed with Spark, yes, it is case-insensitive. Thanks!